### PR TITLE
add vbatts/tar-split review

### PR DESCRIPTION
**Pull request by runcom**
@ingvagabund 

Signed-off-by: Antonio Murdaca runcom@redhat.com

### CODE REVIEW

File: golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec [hunk 1]
```diff
@@ -0,0 +1,177 @@
+%if 0%{?fedora} || 0%{?rhel} == 6
+%global with_devel 1
+%global with_bundled 0
+%global with_debug 1
+%global with_check 0
```
Comment (ingvagabund) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec]: Locally all checks pass. Any reason not to run them?

Comment (runcom) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec]: none, I'll enable them

File: golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+%if ! 0%{?with_bundled}
+BuildRequires: golang(github.com/Sirupsen/logrus)
+BuildRequires: golang(github.com/codegangsta/cli)
+%endif
+
+Provides:      golang(%{import_path}/archive/tar) = %{version}-%{release}
+Provides:      golang(%{import_path}/tar/asm) = %{version}-%{release}
+Provides:      golang(%{import_path}/tar/storage) = %{version}-%{release}
+Provides:      golang(%{import_path}/cmd/tar-split) = %{version}-%{release}
```
Comment (ingvagabund) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec]: cmd/tar-split is not importable. tar-split contains only files with 'package name' directive. So it is not actually provided.

What is missing is golang(%{import_path}/version)

**Comment by ingvagabund** [comment]
Wrt. golang packaging, the rest is fine.

File: golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec (hunk 2 above)
Comment (runcom) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec]: > What is missing is golang(%{import_path}/version)

nope, version should not be imprted nor importable, it's just utility

Comment (runcom) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec]: > cmd/tar-split is not importable. tar-split contains only files with 'package name' directive. So it is not actually provided.

I'll remove it

**Comment by runcom** [comment]
@ingvagabund PTAL I've fixed your comments

File: golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+%if ! 0%{?with_bundled}
+BuildRequires: golang(github.com/Sirupsen/logrus)
+BuildRequires: golang(github.com/codegangsta/cli)
+%endif
+
+Provides:      golang(%{import_path}/archive/tar) = %{version}-%{release}
+Provides:      golang(%{import_path}/tar/asm) = %{version}-%{release}
+Provides:      golang(%{import_path}/tar/storage) = %{version}-%{release}
+
```
Comment (ingvagabund) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec:64]: Missing 
Provides:      golang(%{import_path}/version) = %{version}-%{release}

Comment (runcom) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec:64]: https://github.com/gofed/reviews/pull/1#discussion_r51452256

Comment (ingvagabund) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec:64]: version.go:
package version
// AUTO-GENEREATED. DO NOT EDIT
// 2015-08-14 09:56:50.742727493 -0400 EDT

// VERSION is the generated version from /home/vbatts/src/vb/tar-split/version
var VERSION = "v0.9.6-1-gc76e420"

VERSION symbol is exported. Thus in can be used on other packages. The chance for using the symbol in other projects is very low. Still, it is exported and thus importable.

There is no guideline forcing to provide full list of importable packages. If it is not meant to be importable, it should start with small first letter or it should be at least commented it is not meant to be exported.

Comment (runcom) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec:64]: > it should start with small first letter or it should be at least commented it is not meant to be exported.

right

Comment (runcom) [inline golang-github-vbatts-tar-split/golang-github-vbatts-tar-split.spec:64]: fixed

**Comment by ingvagabund** [comment]
LGTM

**Comment by runcom** [comment]
I'll open another PR since I didn't regenerated the SRPM

**Pull request merged**